libminifi/src/core/logging/WindowsEventLogSink.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef WIN32
+
+#include "core/logging/WindowsEventLogsink.h"
+
+#include "core/logging/WindowsMessageTextFile.h"
+#include "Exception.h"
+
+#include "spdlog/common.h"
+#include "spdlog/sinks/sink.h"
+#include "spdlog/details/log_msg.h"
+
+#include <string>
+
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace core {
+namespace logging {
+namespace internal {
+
+WORD windowseventlog_sink::type_from_level(const spdlog::details::log_msg& msg) const {
+  switch (static_cast<int>(msg.level)) {
```

Comment:
As you said, we don't use spdlog::level::off, it is not well documented either what it should be mapped to, so I've decided to log all unexpected/potential future unknown enums at error log level.

Reply:
Completely agree, I prefer that solution as well, but we should clean up our warnings a bit first, because currently we are not warning free even with our current set of warnings, and any new warning would be most likely lost in the current ones. I wouldn't put this on -Werror (or only just in libminifi), because that could break third party extensions.